### ISSUE TITLE
Transformation is incorrect when offset applied

### ISSUE THREAD

**Issue by bobbyinsane** [konvajs/konva#386]
I'm trying zoom-in zoom-out functionality.

![zoom-transform](https://user-images.githubusercontent.com/982944/39143097-6861afba-474a-11e8-86c8-3ccf45cf3754.gif)

```
// _x_ and _y_ are the initial values when a shape is created
zoom(factor = 1){
    items.forEach(item =>{       
      item.x(item["_x_"] * factor);
      item.y(item["_y_"] * factor);
      item.scaleX(factor);
      item.scaleY(factor);
    });
    layer.draw();
  }
```
if I call above code with transformation already applied (with transform-tool), the shape shifts to a few pixels in both directions. 

I thought it could be because of offset position, so I center it by width/2, height/2. Then I get unusual results. Please check the codepen below.

https://codepen.io/anon/pen/WJrMog

**Comment by mbriggs** [comment]
I am seeing this too. Only when you grab the resize handle and move it, the element spins wildly (reproducible in the code pen)

**Comment by lavrton** [comment]
Fixed and published as `v2.1.3`.
